File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        visited = set()
        result = set() 
        def dfs(node, n):
            if not node:
                result.add(n)
                return 
            if node in visited: return 
            visited.add(node)
            dfs(node.left, n+1)
            dfs(node.right, n+1)
        dfs(root, 0)
        return max(result)  

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> umap;
        int l = 0;
        int r = 0;
        int max = 0; 
        for (int i = 0; i < s.size(); i++) {
            umap[s[i]] = 0;
        }
        for (r = 0; r < s.size(); r++) {
            umap[s[r]] += 1; 
            while (l < r && umap[s[r]] > 1) {
                umap[s[l]] -= 1;
                l++;
            }
            if (r-l+1 > max) max = r-l+1;

        }
        return max;
    }
};
